config/src/main/java/com/typesafe/config/impl/ConfigImplUtil.java

```diff
@@ -233,4 +234,23 @@ static String toCamelCase(String originalName) {
         }
         return nameBuilder.toString();
     }
+
+    /**
+     * Guess configuration syntax from given filename.
+     *
+     * @param filename configuration filename
+     * @return configuration syntax if a match is found. Otherwise, null.
+     */
+    public static ConfigSyntax syntaxFromExtension(String filename) {
+        if (filename == null)
+            return null;
+        if (filename.endsWith(".json"))
```

Comment:
this could be an "else if" I think?

Reply:
sure, I forgot to add that, will do it in a moment